pkg/k8sutil/k8sutil.go

```diff
@@ -83,3 +87,32 @@ func ResourceExists(dc discovery.DiscoveryInterface, apiGroupVersion, kind strin
 	}
 	return false, nil
 }
+
+// GetPod returns a Pod object that corresponds to the pod in which the code
+// is currently running.
+// It expects the environment variable POD_NAME to be set by the downwards API.
+func GetPod(ctx context.Context, client crclient.Client, ns string) (*corev1.Pod, error) {
+	podName := os.Getenv(PodNameEnvVar)
+	if podName == "" {
+		return nil, fmt.Errorf("required env %s not set, please configure downward API", PodNameEnvVar)
+	}
+
+	log.V(1).Info("Found podname", "Pod.Name", podName)
+
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+	}
+	key := crclient.ObjectKey{Namespace: ns, Name: podName}
+	err := client.Get(ctx, key, pod)
+	if err != nil {
+		log.Error(err, "Failed to get Pod currently running in", "Pod.Namespace", ns, "Pod.Name", podName)
+		return nil, err
+	}
+
+	log.V(1).Info("Found Pod currently running in", "Pod.Name", pod.Name)
```

Comment:
Same nit as above. Should also probably include the namespace here too.

```suggestion
	log.V(1).Info("Found Pod", "Pod.Namespace", ns, "Pod.Name", pod.Name)
```